cfme/generic_objects/definition/ui.py

```diff
@@ -0,0 +1,466 @@
+    open_url = BootstrapSwitch(name='open_url')
+    display_for = BootstrapSelect(name='display_for')
+    submit_version = BootstrapSelect(name='submit_how')
+    system_process = BootstrapSelect(name='ae_instance')
+    system_message = Input(name='ae_message')
+    request = Input(name='request')
+    add_attribute_value_key = Button('Add Attribute/Value Pair')
+    attributes = ButtonParameterForm(param_type="Attribute")
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == 'Add a new Custom Button'
+
+
+class GenericObjectButtonGroupAddView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+
+    name = Input(name='name')
+    display = BootstrapSwitch(name='display')
+    description = Input(name='description')
+    image = VersionPick({
+        Version.lowest(): BootstrapSelect('button_image'),
+        '5.9': FonticonPicker('button_icon')})
+
+    add = Button('Add')
+    cancel = Button('Cancel')
+
+    @property
+    def is_displayed(self):
+        return self.title.text == 'GenericObjectButtonGroupAddView'
+
+    def after_fill(self, was_change):
+        # we need to click somewhere out side the form to get add button active,
+        #  after icon is filled
+        if was_change:
+            self.browser.element('//body').click()
+
+
+class GenericObjectButtonGroupDetailsView(GenericObjectDefinitionView):
+    title = Text('#explorer_title_text')
+    configuration = Dropdown(text='Configuration')
+    basic_infornation = SummaryForm('Basic Information')
+    accordion = View.nested(AccordionForm)
+    table = Table('//h3[contains(text(), "Buttons")]/following-sibling::table')
+
+    @property
+    def is_displayed(self):
+        return self.basic_infornation.is_displayed
+
+
+@MiqImplementationContext.external_for(GenericObjectDefinitionCollection.create, ViaUI)
+def create(self, name, description, attributes=None, associations=None, methods=None,
+           custom_image_file_path=None, cancel=False):
+    """
+    Create new generic object definition
+    Args:
+        name: generic object definition name
+        description: generic object definition description
+        attributes:  generic object definition attributes ex. {'address': 'string'}
+        associations:  generic object definition associations ex. {'services': 'Service'}
+        methods: generic object definition methods ex. ['method1', 'method2']
+        custom_image_file_path: generic object definition custom image file path
+        cancel: by default will not be canceled, pass True to make successful cancel
+
+    Returns:
+        GenericObjectDefinition entity
+
+    """
+    view = navigate_to(self, 'Add')
+    view.fill({
+        'name': name,
+        'description': description
+    })
+    view.associations.fill(associations)
+    view.attributes.fill(attributes)
+    view.methods.fill(methods)
+    if custom_image_file_path:
```

Comment:
**REQUIRED**
Please implement `after_fill()` method in `custom_image_file` view  with the following actions.